sled-agent/src/bootstrap/http_entrypoints.rs

```diff
@@ -168,3 +169,23 @@ async fn sled_reset(
     ba.sled_reset().await.map_err(|e| Error::from(e))?;
     Ok(HttpResponseUpdatedNoContent())
 }
+
+#[derive(Debug, Deserialize, JsonSchema)]
+pub struct SledShutdownRequest {
+    /// The reason for the shutdown.
+    pub reason: String,
+}
+
+/// Shuts down the sled, deleting all zones but not user data.
+#[endpoint {
+    method = POST,
+    path = "/sled-shutdown",
+}]
+async fn sled_shutdown(
```

Comment:
Should we be more clear with naming here, and in the methods we call?

The implementation, as you know, doesn't actually shutdown the sled. It puts all control plane state into a "disabled" state. Maybe "sled_agent_shutdown" would be a more apt name?

I just don't want anyone to think this implies anything about the sled as a whole, or about power states.